controllers/useraccount/useraccount_controller.go

```diff
@@ -33,17 +33,23 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
-type IdentityProvider interface {
-	GetIdP() string
-}
-
 // SetupWithManager sets up the controller with the Manager.
 func (r *Reconciler) SetupWithManager(mgr manager.Manager) error {
 	mapToOwnerByLabel := handler.EnqueueRequestsFromMapFunc(commoncontroller.MapToOwnerByLabel("", toolchainv1alpha1.OwnerLabelKey))
+	mapToOwnerByName := handler.EnqueueRequestsFromMapFunc(func(obj client.Object) []reconcile.Request {
+		return []reconcile.Request{
+			{
+				NamespacedName: types.NamespacedName{
+					Namespace: obj.GetNamespace(),
+					Name:      obj.GetName(),
+				},
+			},
+		}
+	})
 
 	return ctrl.NewControllerManagedBy(mgr).
 		For(&toolchainv1alpha1.UserAccount{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
-		Owns(&toolchainv1alpha1.NSTemplateSet{}).
+		Watches(&source.Kind{Type: &toolchainv1alpha1.NSTemplateSet{}}, mapToOwnerByName).
```

Comment:
you don't need a new mapper for this - you can use 
```suggestion
		Watches(&source.Kind{Type: &toolchainv1alpha1.NSTemplateSet{}}, &handler.EnqueueRequestForObject{}).
```

Reply:
ah right, indeed! 🤦‍♂️